crypto/crypto/src/ed25519.rs

```diff
@@ -176,6 +176,16 @@ impl SigningKey for Ed25519PrivateKey {
         let sig = expanded_secret_key.sign(message.as_ref(), &public_key.0);
         Ed25519Signature(sig)
     }
+
+    #[cfg(test)]
+    fn sign_arbitrary_message(&self, message: &[u8]) -> Ed25519Signature {
```

Comment:
You could extract this logic to a private method and reuse it on both `sign_message` and `sign_arbitrary_message`. Same for `multi_ed25519`.

Reply:
Done, note it can't be done with `multi_ed25519` since its `sign_arbitrary_message` can't dispatch to the private `sign_arbitrary_message` of `ed25519`.